moflask/requests/rest.py

```diff
@@ -0,0 +1,48 @@
+"""Helpers for handling REST-APIs."""
+
+import functools
+import urllib.parse
+
+import requests.exceptions
+
+from . import sessions
+
+
+class Client:
+    """A General purpose REST / JSON API-client."""
+
+    def __init__(self, base_url: str, auth=None, request_timeout=2):
+        """Create a new API client instance."""
+        self._session = sessions.Session()
+        self._session.timeout = request_timeout
+        if auth:
+            self._session.auth = auth
```

Comment:
What could `auth` be?

Reply:
The type signature of auth would be `Callable[[requests.PreparedRequest], requests.PreparedRequest]`. So the simplest one would be:

```
def auth_middleware(request):
    return request
```